src/app/Dashboard/Charts/ChartCard.tsx

```diff
@@ -0,0 +1,360 @@
+
+  const refresh = React.useCallback(() => {
+    controllerContext.controller.requestRefresh();
+  }, [controllerContext]);
+
+  React.useEffect(() => {
+    refresh();
+    addSubscription(interval(props.period * 1000).subscribe(() => refresh()));
+  }, [addSubscription, props.period, refresh]);
+
+  const popout = React.useCallback(() => {
+    if (chartSrc && dashboardUrl) {
+      window.open(chartSrc, '_blank');
+    }
+  }, [chartSrc, dashboardUrl]);
+
+  const cardStyle = React.useMemo(() => {
+    return {
+      height: getHeight(props.chartKind),
+    };
+  }, [props.chartKind]);
+
+  const expandButton = React.useMemo(() => {
+    return (
+      <>
+        <Button
+          key={0}
+          aria-label={`Expand ${props.chartKind} chart window`}
+          onClick={updateKey}
+          variant="plain"
+          icon={<SyncAltIcon />}
+          isDisabled={!chartSrc || !dashboardUrl}
+        />
+      </>
+    );
+  }, [props.chartKind, updateKey, chartSrc, dashboardUrl]);
+
+  const popoutButton = React.useMemo(() => {
+    return (
+      <>
+        <Button
+          key={1}
+          aria-label={`Pop out ${props.chartKind} chart`}
+          onClick={popout}
+          variant="plain"
+          icon={<ExternalLinkAltIcon />}
+          isDisabled={!chartSrc || !dashboardUrl}
+        />
+      </>
+    );
+  }, [props.chartKind, popout, chartSrc, dashboardUrl]);
+
+  const actions = React.useMemo(() => {
+    const a = props.actions || [];
+    return [expandButton, popoutButton, ...a];
+  }, [props.actions, expandButton, popoutButton]);
+
+  const header = React.useMemo(() => {
+    const isWide = props.span > 4;
+    const style = {
+      marginBottom: isWide ? '-2.8em' : '',
+    };
+    if (hasRecording) {
+      return (
+        <>
+          <CardHeader style={style}>
+            <CardActions>{actions}</CardActions>
+          </CardHeader>
+        </>
+      );
+    } else {
+      return (
+        <>
+          <CardHeader style={style}>
+            <CardTitle>{props.chartKind}</CardTitle>
+            <CardActions>{props.actions}</CardActions>
+          </CardHeader>
+        </>
+      );
+    }
+  }, [props.actions, props.span, props.chartKind, hasRecording, actions]);
+
+  const handleCreateRecording = React.useCallback(() => {
+    history.push({
+      pathname: '/recordings/create',
+      state: {
+        restartExisting: true,
+        name: RECORDING_NAME,
+        templateName: 'Profiling',
+        templateType: 'TARGET',
+        labels: [{ key: 'origin', value: RECORDING_NAME }],
+        duration: -1,
+        // TODO these are arbitrary defaults that will be set in the recording creation form.
+        // Should these values be inferred in some more intelligent way?
+        maxAge: 120, // seconds
+        maxSize: 100 * 1024 * 1024, // bytes
+      } as CreateRecordingProps,
+    });
+  }, [history]);
+
+  return (
+    <>
+      <DashboardCard
+        key={idx}
```

Comment:
Not really sure why, but having this key re-renders the card every time you resize the card...